src/backend/distributed/utils/multi_partitioning_utils.c

```diff
@@ -40,9 +40,15 @@ static char * PartitionBound(Oid partitionId);
 bool
 PartitionedTable(Oid relationId)
 {
-	Relation rel = heap_open(relationId, AccessShareLock);
+	Relation rel = try_relation_open(relationId, AccessShareLock);
 	bool partitionedTable = false;
 
+	/* don't error out for tables that are dropped */
+	if (rel == NULL)
```

Comment:
I couldn't see the reason that we're doing this for. How could this change lead to getting a NULL relation here?